src/main/java/de/tum/in/www1/artemis/service/connectors/athena/AthenaSubmissionSelectionService.java

```diff
@@ -0,0 +1,113 @@
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.RestTemplate;
+
+import de.tum.in.www1.artemis.domain.TextExercise;
+import de.tum.in.www1.artemis.domain.TextSubmission;
+import de.tum.in.www1.artemis.exception.NetworkingError;
+import de.tum.in.www1.artemis.service.dto.athena.TextExerciseDTO;
+
+/**
+ * Service for selecting the "best" submission to assess right now using Athena, e.g. by the highest information gain.
+ * Assumes that submissions have already been sent to Athena (it only sends submission IDs to choose from).
+ * The default choice if Athena does not respond is to choose a random submission.
+ */
+@Service
+@Profile("athena")
+public class AthenaSubmissionSelectionService {
+
+    // pretty short timeout, because this should be fast, and it's not too bad if it fails
+    private static final int REQUEST_TIMEOUT_MS = 1000;
+
+    private final Logger log = LoggerFactory.getLogger(AthenaSubmissionSelectionService.class);
+
+    @Value("${artemis.athena.url}")
+    private String athenaUrl;
+
+    private final AthenaConnector<RequestDTO, ResponseDTO> connector;
+
+    private static class RequestDTO {
+
+        public TextExerciseDTO exercise;
+
+        public List<Long> submissionIds; // Athena just needs submission IDs => quicker request, because less data is sent
+
+        RequestDTO(@NotNull TextExercise exercise, @NotNull List<Long> submissionIds) {
+            this.exercise = TextExerciseDTO.of(exercise);
+            this.submissionIds = submissionIds;
+        }
+    }
+
+    private record ResponseDTO(long data // will contain the submission ID to choose, or -1 if no submission was explicitly chosen
+    ) {
+    }
+
+    /**
+     * Create a new AthenaSubmissionSelectionService, which uses a custom timeout for requests to Athena
+     */
+    public AthenaSubmissionSelectionService(@Qualifier("athenaRestTemplate") RestTemplate athenaRestTemplate) {
+        // Configure rest template to use the given timeout
+        SimpleClientHttpRequestFactory requestFactory = new SimpleClientHttpRequestFactory();
+        requestFactory.setConnectTimeout(REQUEST_TIMEOUT_MS);
+        requestFactory.setReadTimeout(REQUEST_TIMEOUT_MS);
+        athenaRestTemplate.setRequestFactory(requestFactory);
+        // Create connector
+        connector = new AthenaConnector<>(log, athenaRestTemplate, ResponseDTO.class);
+    }
+
+    /**
+     * Fetches the proposedTextSubmission for a given exercise from Athena
+     *
+     * @param exercise    the exercise to get the proposed Submission for
+     * @param submissions assessable submissions of the exercise
+     * @return a Submission suggested by the Athena submission selector (e.g. chosen by the highest information gain)
+     * @throws IllegalArgumentException if exercise isn't automatically assessable
+     */
+    public Optional<TextSubmission> getProposedSubmission(TextExercise exercise, List<TextSubmission> submissions) {
```

Comment:
Do you really need all submissions (objects) here? I think it would be better if you just use the ids here and then load only the one selected submission from the database. Keep in mind that we have courses with up to 2000 students, so a list of 2000 text submission would be needed here.

Reply:
It makes sense to split it up 👍  Somehow, the old Athena worked despite always sending all submissions at once, but this solution is better - so I will change that.

Reply:
Good input! I changed this function to get submission IDs. However, I kept the submission objects in `TextSubmissionService.getRandomTextSubmissionEligibleForNewAssessment`. There are two reasons for this:
1. The logic used in `SubmissionService.getAssessableSubmissions` is quite complex and hard to get in a pure database call, so I'm keeping it as-is.
2. With my latest changes in https://github.com/ls1intum/Artemis/pull/6861/commits/9c33e20d6dda53b8b958504c01dd5018444e929d, the memory usage should be as before (loading the submissions exactly once). To improve this, one would have to refactor quite a lot (e.g., using streams only instead of lists?). I don't want to do that in this PR, if possible, as it's quite large by itself already.

Reply:
Good input! I changed this function to get submission IDs. However, I kept the submission objects in `TextSubmissionService.getRandomTextSubmissionEligibleForNewAssessment`. There are two reasons for this:
1. The logic used in `SubmissionService.getAssessableSubmissions` is quite complex and hard to get in a pure database call, so I'm keeping it as-is.
2. With my latest changes in https://github.com/ls1intum/Artemis/pull/6861/commits/9c33e20d6dda53b8b958504c01dd5018444e929d, the memory usage should be as before (loading the submissions exactly once). To improve this, one would have to refactor quite a lot (e.g., using streams only instead of lists?). I don't want to do that in this PR, if possible, as it's quite large by itself already.